### PR TITLE
Fix/move cc files into src folder

### PR DESCRIPTION
**Pull request by OHerasym**
PR Change:
- Move cc files to src directory
- Build binary in bin folder
- Add copying xml interfaces before starting SDL
- Add pathToSDL & pathToSDLInterfaces checks

Related: [APPLINK-27862](https://adc.luxoft.com/jira/browse/APPLINK-27862)
               [APPLINK-27864](https://adc.luxoft.com/jira/browse/APPLINK-27864)

### CODE REVIEW

**Comment by OHerasym** [comment]
Please review: @AGritsevich, @anosach-luxoft , @VProdanov, @VVeremjova

**Comment by OHerasym** [comment]
Please review: @vlantonov

File: modules/atf/util.lua [hunk 1]
```diff
@@ -63,6 +63,19 @@ function table.removeKey(t, k)
...
 end
 
+function check_required_fields()
+  if (not is_file_exists(config.pathToSDL.."smartDeviceLinkCore")) and 
+     (not is_file_exists(config.pathToSDL.."/smartDeviceLinkCore")) then
+    print("ERROR: SDL is not accessible at the specified path: "..config.pathToSDL)
+    os.exit(1)
+  end
+  if (not is_file_exists(config.pathToSDLInterfaces.."MOBILE_API.xml")) and 
+     (not is_file_exists(config.pathToSDLInterfaces.."/MOBILE_API.xml")) then
```
Comment (vlantonov) [inline modules/atf/util.lua]: @OHerasym Shouldn't there be a check for the existence of `HMI_API.xml`?

Comment (OHerasym) [inline modules/atf/util.lua]: @vlantonov , This check is for correct path to *.xml interfaces, to omit errors with wrong or empty path. Actually I can add check for HMI_API, but I don't think that is necessary

Comment (vlantonov) [inline modules/atf/util.lua]: @OHerasym OK

**Pull request merged**